mozjs/js/rust/src/magicdom/htmlelement.rs

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Wlic
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use jsapi::root::*;
+use conversions::{ConversionResult, ConversionBehavior, FromJSValConvertible, ToJSValConvertible};
+use glue::CreateCallArgsFromVp;
+extern crate libc;
+
+magic_dom! {
+    HtmlElement,
+    HTMLELEMENT_CLASS,
+    HtmlElement_constructor,
+    magic_dom_spec_HtmlElement,
+    struct HtmlElement_spec {
+        // TODO put Element here
```

Comment:
Note: this isn't quite what we want to do. We want to start these fields' slot offsets *after* all of `Element`'s slots, and when we create an `HtmlElement` object, we want its prototype chain to reflect that it inherits from `Element`.